### PR TITLE
[TRAFFIC-9353][TRAFFIC-9354][TRAFFIC-9355]: Add `confluent network private-link attachment update|delete|create` commands

### PR DESCRIPTION
**Pull request by syoung-cflt**
What
----

Note that this is merging into the feature branch https://github.com/confluentinc/cli/tree/traffic-8850-network-cli. We will merge commits into main from this feature branch after all of the commands for network are implemented

* Add `confluent network private-link attachment update`
* Add `confluent network private-link attachment delete`
* Add `confluent network private-link attachment create`

References
----------

[CLI Commands Design](https://confluentinc.atlassian.net/wiki/spaces/PM/pages/2987394300/1-pager+--+Network+CLI)
[Implementation 1-Pager](https://confluentinc.atlassian.net/wiki/spaces/~712020463029eabbac401bbe1b033ce0d3a00e/pages/3136749631/Network+CLI)

Test & Review
-------------

Added integration tests.

Performed manual tests

Open Questions / Follow-ups
---------------------------

Note that this "private link attachment" feature is tied to ESKU which is only available in AWS at the moment. I confirmed with our PM @kabraham13 that once we support ESKU in other clouds, the engineers who work on the API changes for that project should also be responsible for updating the CLI.

### CODE REVIEW

File: internal/network/command_private_link_attachment_delete.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+	deleteFunc := func(id string) error {
+		if err := c.V2Client.DeletePrivateLinkAttachment(environmentId, id); err != nil {
+			return errors.Errorf(errors.DeleteResourceErrorMsg, resource.PrivateLinkAttachment, id, err)
+		}
+		return nil
+	}
+
+	deletedIds, err := deletion.DeleteWithoutMessage(args, deleteFunc)
+	deleteMsg := "Requested to delete %s %s.\n"
```
Comment (brianstrauch) [inline internal/network/command_private_link_attachment_delete.go:55]: We already have `errors.RequestedDeleteResourceMsg`

File: internal/network/command_private_link_attachment_update.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+func (c *command) newPrivateLinkAttachmentUpdateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:               "update <id>",
+		Short:             "Update an existing private link attachment.",
+		Args:              cobra.ExactArgs(1),
+		ValidArgsFunction: pcmd.NewValidArgsFunction(c.validPrivateLinkAttachmentArgs),
+		RunE:              c.privateLinkAttachmentUpdate,
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: `Update the name of private link attachment "platt-123456"`,
```
[the remark this answers is not in the document]
Comment (syoung-cflt) [reply internal/network/command_private_link_attachment_update.go:21]: Thanks! I will update the missing `.` in another files as well.

File: internal/network/command_private_link_attachment_create.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "Create an AWS PrivateLink attachment.",
+				Code: "confluent network private-link attachment create aws-private-link-attachment --cloud aws --region us-west-2",
+			},
+		),
+	}
+
+	pcmd.AddCloudFlag(cmd)
+	cmd.Flags().String("region", "", "Cloud service provider region where the resources to be accessed using the PrivateLink attachment are located.")
```
[the remark this answers is not in the document]
Comment (syoung-cflt) [reply internal/network/command_private_link_attachment_create.go:29]: Yeah this one should be "private link".

File: internal/network/command_private_link_attachment_delete.go (hunk 1 above)
Comment (syoung-cflt) [reply internal/network/command_private_link_attachment_delete.go:55]: nice, i will update other files as well

File: internal/network/command_private_link_attachment_create.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (syoung-cflt) [reply internal/network/command_private_link_attachment_create.go:29]: I will also update other files.
For AWS, it should be "AWS PrivateLink".
For Azure, it should be "Azure Private Link".
For GCP, it should be "GCP Private Service Connect"

**Review by ejagade — APPROVED** [review]

File: internal/network/command_private_link_attachment_delete.go (hunk 1 above)
Comment (syoung-cflt) [reply internal/network/command_private_link_attachment_delete.go:55]: @brianstrauch it looks like the `errors.RequestedDeleteResourceMsg` [here](https://github.com/confluentinc/cli/blob/main/pkg/errors/strings.go#L115) is `"Requested to delete %s \"%s\".\n"` which is not compatible with `utils.ArrayToCommaDelimitedString(deletedIds, "and")` [here](https://github.com/confluentinc/cli/blob/ab8604ce15331fa582cd944ec3eadacd9740ae96/pkg/utils/utils.go#L104).

There are extra double quotes in the output.

File: internal/network/command_private_link_attachment_create.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (syoung-cflt) [reply internal/network/command_private_link_attachment_create.go]: Updated!

File: internal/network/command_private_link_attachment_update.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (syoung-cflt) [reply internal/network/command_private_link_attachment_update.go]: Fixed!

File: internal/network/command_private_link_attachment_delete.go (hunk 1 above)
Comment (brianstrauch) [reply internal/network/command_private_link_attachment_delete.go:55]: Good point, let's use a custom format string!

**Review by brianstrauch — APPROVED** [review]

**Pull request merged**